lib/rubocop/cop/rspec/be_eq.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for expectations where `be(...)` can replace `eq(...)`.
+      #
+      # The `be` matcher compares by identity while the `eq` matcher compares
+      # using `==`. Booleans and nil can be compared by identity and therefore
+      # the `be` matcher is preferable as it is a more strict test.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to eq(true)
+      #   expect(foo).to eq(false)
+      #   expect(foo).to eq(nil)
+      #
+      #   # good
+      #   expect(foo).to be(true)
+      #   expect(foo).to be(false)
+      #   expect(foo).to be(nil)
+      #
+      class BeEq < Base
+        extend AutoCorrector
+
+        MSG = 'Prefer `be` over `eq`.'
+        RESTRICT_ON_SEND = %i[to not_to to_not].freeze
+
+        # @!method eq_type_with_identity(node)
+        def_node_matcher :eq_type_with_identity, <<-PATTERN
+          (send _ #Runners.all $(send nil? :eq {true false nil_type?}))
```

Comment:
I just wonder why in the BeNil cop you opted for just `(send nil? :be nil)` and here you're checking that it's being passed to a runner (to/not_to)

Reply:
I stole it from [the BeEql cop](https://github.com/rubocop/rubocop-rspec/blob/766bc1b30a17ade4d36cee8dd310c71e66d5e8a9/lib/rubocop/cop/rspec/be_eql.rb#L44-L47), which applies it only `to` and specifically not to `not_to` and `to_not`. Since the `BeEq` cop can (in my opinion) apply also to negative matchers, maybe I should just change the pattern to `(send nil? :eq {true false nil_type?})`?

Reply:
Fixed, squashed and pushed.